sympy/physics/continuum_mechanics/truss.py

```diff
@@ -0,0 +1,598 @@
+                            self._nodes_occupied[(new_label, member[2])] = True
+                            self._nodes_occupied[(member[2], new_label)] = True
+                            self._nodes_occupied.pop(tuple([label, member[2]]))
+                            self._nodes_occupied.pop(tuple([member[2], label]))
+                        elif member[2] == node[0]:
+                            member[2] = new_label
+                            self._member_nodes[member[0]] = [member[1], new_label]
+                            self._nodes_occupied[(member[1], new_label)] = True
+                            self._nodes_occupied[(new_label, member[1])] = True
+                            self._nodes_occupied.pop(tuple([member[1], label]))
+                            self._nodes_occupied.pop(tuple([label, member[1]]))
+                    self._nodes[self._nodes.index((label, node[1], node[2]))] = (new_label, node[1], node[2])
+                    self._node_labels[self._node_labels.index(node[0])] = new_label
+                    self._loads[new_label] = self._loads[node[0]]
+                    self._loads.pop(node[0])
+                    self._supports[new_label] = self._supports[node[0]]
+                    self._supports.pop(node[0])
+
+    def change_member_label(self, label, new_label):
+        """
+        This method changes the label of a member.
+
+        Parameters
+        ==========
+        label: String or Symbol
+            The label of the member for which the label has
+            to be changed.
+
+        new_label: String or Symbol
+            The new label of the member.
+
+        Examples
+        ========
+
+        >>> from sympy.physics.continuum_mechanics.truss import Truss
+        >>> t = Truss()
+        >>> t.add_node('A', 0, 0)
+        >>> t.add_node('B', 3, 0)
+        >>> t.nodes
+        [('A', 0, 0), ('B', 3, 0)]
+        >>> t.change_node_label('A', 'C')
+        >>> t.nodes
+        [('C', 0, 0), ('B', 3, 0)]
+        >>> t.add_member('BC', 'B', 'C')
+        >>> t.members
+        [['BC', 'B', 'C']]
+        >>> t.change_member_label('BC', 'BC_new')
+        >>> t.members
+        [['BC_new', 'B', 'C']]
+        """
+        if label not in self._member_labels:
+            raise ValueError("No such member exists for the Truss")
+
+        else:
+            members_duplicate = self._members.copy()
+            for member in members_duplicate:
+                if member[0] == label:
+                    self._member_labels[self.member_labels.index(member[0])] = new_label
+                    self._member_nodes[new_label] = [self._member_nodes[label][0], self._member_nodes[label][1]]
+                    self._member_nodes.pop(label)
+                    self._internal_forces[new_label] = self._internal_forces[label]
+                    self._internal_forces.pop(label)
+                    self._members[self._members.index([label, member[1], member[2]])] = [new_label, member[1], member[2]]
```

Comment:
Why do we have three member data structures? 
```
self._member_labels
self._members
self._member_nodes
```
Are they really required?
If not, you can just keep `self._members` in the form of `self._member_nodes` ( a dictionary with member label as key and (node1,node2) as values.